apps/spruce/src/pages/task/taskTabs/TaskHistory/index.tsx

```diff
@@ -78,17 +80,43 @@ const TaskHistory: React.FC<TaskHistoryProps> = ({ task }) => {
   });
 
   const { taskHistory } = data ?? {};
-  const { tasks = [] } = taskHistory ?? {};
+  const { pagination, tasks = [] } = taskHistory ?? {};
+  const { mostRecentTaskOrder, oldestTaskOrder } = pagination ?? {};
 
   const groupedTasks = groupTasks(tasks, shouldCollapse);
   const numVisibleTasks = Math.floor(timelineWidth / SQUARE_WITH_BORDER);
-  const visibleTasks = groupedTasks.slice(0, numVisibleTasks);
+
+  const visibleTasks =
```

Comment:
The number of visible tasks should be the next limit for the history query.

Reply:
I think that it's safer to request 50 in case the user decides to expand the page (the content will look really short otherwise since we don't re-request on page resize events). It could also be an improvement made in a future ticket

Reply:
oh yeah no data should be missing... if it's missing that is a bug. Let me know if you find any issues with missing data!

Reply:
I tested this a bunch more and found that the taskbox indices are preserved when paginating regardless of page width!